sdk/python/lib/pulumi/invoke.py

```diff
@@ -127,3 +134,70 @@ def merge(
         dest.version = dest.version if source.version is None else source.version
 
         return dest
+
+
+class InvokeTransformArgs:
+    """
+    InvokeTransformArgs is the argument bag passed to a resource transform.
+    """
+
+    token: str
+    """
+    The token of the Invoke.
+    """
+
+    args: "Inputs"
+    """
+    The original args passed to the Invoke call.
+    """
+
+    opts: "InvokeOptions"
+    """
+    The original invoke options passed to the Invoke call.
+    """
+
+    def __init__(
+        self,
+        token: str,
+        args: "Inputs",
+        opts: "InvokeOptions",
+    ) -> None:
+        self.token = token
+        self.args = args
+        self.opts = opts
+
+
+class InvokeTransformResult:
+    """
+    InvokeTransformResult is the result that must be returned by an invoke transform callback.
+    It includes new values to use for the `args` and `opts` of the `Invoke` in place of the
+    originally provided values.
+    """
+
+    args: "Inputs"
+    """
+    The new args to use in place of the original `args`.
+    """
+
+    opts: "InvokeOptions"
+    """
+    The new invoke options to use in place of the original `opts`
```

Comment:
Think we should be consistent with/with absence of periods at end of doc comments.